Framework/Algorithms/src/Q1DWeighted.cpp

```diff
@@ -228,10 +244,22 @@ void Q1DWeighted::exec() {
         // For reference - in the case where we don't use sub-pixels, simply
         // use:
         //     double sinTheta = sin( spectrumInfo.twoTheta(i)/2.0 );
+        const double wavelength = (XIn[j] + XIn[j + 1]) / 2.;
         V3D pos = spectrumInfo.position(i) - V3D(subX, subY, 0.0) - samplePos;
-        double sinTheta = sin(0.5 * pos.angle(beamLine));
-        double factor = fmp * sinTheta;
-        double q = factor * 2.0 / (XIn[j] + XIn[j + 1]);
+
+        if (correctGravity) {
+          const double l2 = spectrumInfo.l2(i);
+          const double tof =
+              UnitConversion::run("Wavelength", "TOF", wavelength, l1, l2, 0.,
+                                  DeltaEMode::Elastic, 0.) *
+              1E-6 * l2 / (l1 + l2);
+          const double drop = PhysicalConstants::g * tof * tof / 2.;
+          V3D correction(0., drop, 0.);
+          pos += correction;
+        }
```

Comment:
What is the reasons for not using https://github.com/mantidproject/mantid/blob/master/Framework/Algorithms/inc/MantidAlgorithms/GravitySANSHelper.h here?

Reply:
Good point, I will have a look. 👍